src/main/java/org/mapstruct/intellij/inspection/UnmappedTargetPropertiesInspection.java

```diff
@@ -181,6 +190,14 @@ private static boolean isBeanMappingIgnoreByDefault(PsiMethod method) {
             return false;
         }
 
+        private static boolean isPrototypeMethod(@NotNull PsiMethod method) {
+
+            return Optional.ofNullable( method.getContainingClass() )
+                .map( containingClass -> findAnnotation( containingClass, MAPPER_CONFIG_ANNOTATION_FQN ) )
+                .stream().findFirst()
+                .isPresent();
```

Comment:
Can't this be replaced with `findMapperConfigClass( method.getContainingClass() ).isPresent()`

Reply:
Good idea, but actually not. `findMapperConfigClass(@NotNull PsiAnnotation mapperAnnotation)` takes the mapper annotation as parameter, finds the declared attribute of `@Mapper( config = ... )` and resolves that class.

In this case the prototype-method check searches if the containing class is annotated with `@MapperConfig`:

```java
@MapperConfig
public interface MyConfig {

  // these are allowed to be incomplete, therefore disable the inspection
  Target map(Source source);

}
```

I could make this more expressive by using another method like this (or rename it):

```java
        private static boolean isPrototypeMethod(@NotNull PsiMethod method) {

            return isContainedInsideMapperConfig( method );
        }

        private static boolean isContainedInsideMapperConfig(@NotNull PsiMethod method) {

            return Optional.ofNullable( method.getContainingClass() )
                .map( containingClass -> findAnnotation( containingClass, MAPPER_CONFIG_ANNOTATION_FQN ) )
                .stream().findFirst()
                .isPresent();
        }
```

Reply:
You are indeed right. Looking at our codebase I think that we already have something in `MapStuctUtil` see `MapStructUtil#isMapperConfig`. 

In any case, I would propose an alternative solution. Instead of checking if a method is prototype method. We should check if a method is in a mapper instead (see `MapStructUtil#isMapper`). There are people for some reason that use `@Mapper` and `@MapperConfig` on the same interface. We can do this check even before we get the target type.

Reply:
Very good. I changed it as you proposed. This also avoids some unnecessary compuations.